lsr-role2collection.py

```diff
@@ -329,23 +335,27 @@ def recursive_grep(path, find, file_patterns):
     return False
 
 
-def replace_rolename_with_collection(path, role, collection):
+def replace_rolename_with_collection(path, role):
     """
     Replace the roles or include_role values, ROLE or `linux-system-roles.ROLE`, are replaced
     with `NAMESPACE.COLLECTION.ROLE` in the given dir `path` recursively.
     """
     find = r"( *name: | *- name: | *- | *roletoinclude: | *role: | *- role: )(linux-system-roles\.{0}\b)".format(role, role)
-    replace = r"\1" + namespace + "." + collection + "." + role
+    replace = r"\1" + role
     file_patterns = ['*.yml', '*.md']
     file_replace(path, find, replace, file_patterns)
 
+    find = r"([ -] hosts: [\w\d\.]*\n)"
+    replace = r"\1  collections:\n    - {0}.{1}\n".format(namespace, collection)
```

Comment:
Do we need to use the `collections:` keyword?  Is there some reason we can't just use `include_role: name: fedora.system_roles.ROLENAME` everywhere?

Reply:
The only case I can think of is when a test playbook uses a module/plugin e.g. using `blivet:` in a storage test probably won't work unless the test playbook is inside the collection.  I think you would have to use something like `fedora.system_roles.storage.blivet:` in that case.

Reply:
> Do we need to use the `collections:` keyword? Is there some reason we can't just use `include_role: name: fedora.system_roles.ROLENAME` everywhere?

No need to use the `collections:` keyword. I just thought we as a team are hoping to do so! I can revert the last change.

The same replacement is made on the example playbooks in docs/ROLE (and possibly, example codes in README). Do we want to revert it, as well? Or it is needed for them?

Reply:
>  I just thought we as a team are hoping to do so!

We're following the guidance from Ansible - https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-collections-in-a-playbook
"Once installed, you can reference a collection content by its fully qualified collection name (FQCN):"
"To avoid a lot of typing, you can use the collections keyword added in Ansible 2.8: .... Notice that you still need the FQCN for non-action or module plugins."

And we've been told by the Ansible folks on several occasions that it is better to use the FQCN.

> The same replacement is made on the example playbooks in docs/ROLE ...

I'm still not sure what, if anything, we should say to users about how to consume these roles if using collections . . . I think the latest guidance that we discussed is that we should not mention galaxy or `collections:`, and just use the FQCN in our DOCS/examples - @pcahyna is this your understanding?

Reply:
I think we should prefer using the FQCN and not having to add `collections:` to the playbooks.